napari/layers/image/image.py

```diff
@@ -713,11 +716,14 @@ def _set_view_slice(self) -> None:
         """Set the slice output based on this layer's current state."""
         # Skip if any non-displayed data indices are out of bounds.
         # This can happen when slicing layers with different extents.
-        indices = self._slice_indices
+        data_slice = self._data_slice
         for d in self._slice_input.not_displayed:
-            if (indices[d] < 0) or (indices[d] > self._extent_data[1][d]):
+            pt = data_slice.point[d]
+            low = pt - data_slice.margin_left[d]
+            high = pt + data_slice.margin_right[d]
+            if pt < max(0, low) or pt > min(self._extent_data[1][d], high):
                 self._slice = _ImageSliceResponse.make_empty(
```

Comment:
double check this code (should allow point to be outside of data if margin gets back in it)

Reply:
Fixed this logic and moved it to the `_ImageSliceRequest` class